File: plus-one/plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {
        
        int carry = 1;
        int remainder = 0;
        for(int i = digits.length - 1;i >= 0;i--){
            int num = digits[i];
            num += carry + remainder;
            carry = num/10;
            if(num > 9) num = num % 10;
            digits[i] = num;
        }
        
        if(carry > 0){
            int[] carryArray = new int[digits.length+1];
            carryArray[0] = carry;
            int leftPos = 1;
            for(int i = 0; i < digits.length;i++){
                carryArray[leftPos] = digits[i];
                leftPos++;
            }
            return carryArray;
        }
        
        return digits;
        
        
    }
}

File: plus-one/plus-one.kt
class Solution {
    fun plusOne(digits: IntArray): IntArray {
        
        var carry = 1
        for(i in digits.size-1 downTo 0){
            
            val current = digits[i] 
            val sum = current + carry
            carry = sum / 10
            digits[i] = sum % 10
            
        }
        
        if(carry > 0){
            val increasedArray = IntArray(digits.size + 1)
            increasedArray[0] = carry
            for(i in 0 until digits.size){
                increasedArray[i+1] = digits[i]
            }
            
            return increasedArray
        }
        
        return digits
        
    }
}